scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java

```diff
@@ -472,7 +473,8 @@ protected NodeSet getRMNodes(Map<JobId, JobDescriptor> jobMap, int neededResourc
      */
     private Map<String, Serializable> createBindingsForSelectionScripts(InternalJob job, EligibleTaskDescriptor etd, InternalTask task) throws IOException, ClassNotFoundException {
         Map<String, Serializable> bindings = new HashMap<>();
-        Map<String, Serializable> variables = new HashMap<>();
+        VariablesMap variables = new VariablesMap();
```

Comment:
I don't understand the need to use the VariablesMap in the SchedulingMethodImpl class. There will be no variable propagation involved at this point, so why not using a standard hashmap ?